DirectEmployers/MyJobs
# Pd 3519 assessment record list filter component

## What's this PR do?
Adds working filters, sorting and also the StarDisplay component.
The filters are currently single-select, pending the development of our custom multi-select component. However, all of the code is written to support multi-select filtering. Would just need to connect the new component after it's ready...

## Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/29709626/28673198-f44abfa0-72af-11e7-9673-a897ad46282f.png)


## Where should the reviewer start?
1. RecordListFilter.jsx
2. mainActions.js
3. mainReducer.js
4. RecordListHeader.jsx
5. StarDisplay.jsx

## How should this be manually tested?
Navigate to a partner assessment and try filtering and sorting the data. Make sure that the Stars/score is showing up.

## Any background context you want to provide?
Using mock data in RecordListFilter.jsx:8-17. I'll remove it after we have API endpoints to populate the filters with.
